producer/writer/message.go

```diff
@@ -33,6 +35,7 @@ type message struct {
 
 	pb           msgpb.Message
 	meta         metadata
+	initTime     time.Time
```

Comment:
Perhaps more efficient to just store `initNanos` here?

Reply:
I started with that, but then I have to cast it back to time when reporting, so it's more efficient to use time here

Reply:
You can simply get the nanoseconds from the end time and compute the nanosecond difference. Up to you, but feels inconsistent to me that you have both `xxTime` and `xxNanos` in the `message` struct.

Reply:
right, yeah sg